### PR TITLE
#1196 correct mark as paid flow

### PR DESCRIPTION
**Pull request by gouravmpk**
## Describe your changes
UI changes
## Issue ticket number and link
#1196
## Type of change

UI changes added some conditional rendering and some state management changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have tested on Chrome and Firefox
- [x] I have tested on a mobile device
- [x] I have provided a screenshot or recording of changes in my PR if there were updates to the frontend

## Video
Owner View

https://github.com/stakwork/sphinx-tribes/assets/20900042/20e7c779-f2a3-451f-b1d1-c6eb4eeaf01c

Organitation view with manage bounties access

https://github.com/stakwork/sphinx-tribes/assets/20900042/3ba96dba-619a-4b6e-8424-af73f7926ff0

Note: changes in CodingMobile.tsx line 164 code was setting paid to null which was beaking CodingMobile.tsx button rendering

@ecurrencyhodler @kevkevinpal

### CODE REVIEW

File: frontend/app/src/people/widgetViews/summaries/wantedSummaries/CodingMobile.tsx [hunk 1]
```diff
@@ -161,8 +161,7 @@ export default function MobileView(props: CodingViewProps) {
             />
           </ButtonRow>
 
-          {!!paid ? markUnpaid : null}
-
+          {!!paid ? markUnpaid : markUnpaid}
```
Comment (kevkevinpal) [inline frontend/app/src/people/widgetViews/summaries/wantedSummaries/CodingMobile.tsx:164]: this doesnt really make sense we should just change the prop name to `markPaidOrUnpaid` and then just display it at all times

File: frontend/app/src/people/widgetViews/summaries/wantedSummaries/CodingBounty.tsx [hunk 2]
```diff
@@ -382,31 +389,33 @@ function MobileView(props: CodingBountiesProps) {
...
+            <IconButton
+              width={'100%'}
+              height={48}
+              style={{
+                bottom: '10px',
+                border: `1px solid ${color.primaryColor.P400}`,
+                background: paidStatus ? color.green1 : color.pureWhite,
+                color: paidStatus ? color.white100 : color.borderGreen1
+              }}
+              text={paidStatus ? 'Mark as Unpaid' : 'Mark as Paid'}
```
Comment (kevkevinpal) [inline frontend/app/src/people/widgetViews/summaries/wantedSummaries/CodingBounty.tsx:402]: lets make the text `Mark as Unpaid` and `Mark as Paid` a variable and then use them in the different places so if we decide to change the text it easier to change everywhere

**Comment by kevkevinpal** [comment]
if you can also bump the percentage in `jest.config.json` so the jest test action passes since we'll add the coverage for the file in a different issue that would be good

`Jest: "./src/people/" coverage threshold for lines (7.66%) not met: 7.65%`

**Comment by kevkevinpal** [comment]
> if you can also bump the percentage in `jest.config.json` so the jest test action passes since we'll add the coverage for the file in a different issue that would be good
> 
> `Jest: "./src/people/" coverage threshold for lines (7.66%) not met: 7.65%`

rebased to [3a4c388](https://github.com/stakwork/sphinx-tribes/pull/1218/commits/3a4c3885cdc9d144e9c5aee4f91e20768f71dbad) this adds the test added in https://github.com/stakwork/sphinx-tribes/pull/1219

**Comment by gouravmpk** [comment]
Okay I will

**Comment by gouravmpk** [comment]
@kevkevinpal can I get a quick review

**Comment by ecurrencyhodler** [comment]
@gouravmpk can you post another video of your latest?

**Comment by gouravmpk** [comment]
@ecurrencyhodler Yep

> @gouravmpk can you post another video of your latest?

**Comment by gouravmpk** [comment]
@ecurrencyhodler 
https://github.com/stakwork/sphinx-tribes/assets/20900042/06db9135-40f4-4703-9f07-0cf313c77db4

**Comment by ecurrencyhodler** [comment]
This is still missing the original "mark as paid" flow @gouravmpk. 

Here is a video:
https://github.com/stakwork/sphinx-tribes/assets/32662508/64d444a6-ecad-4be4-9cc9-634c51489f7a

The states look good to me.

File: frontend/app/src/people/widgetViews/summaries/wantedSummaries/CodingMobile.tsx [hunk 3]
```diff
@@ -161,8 +161,7 @@ export default function MobileView(props: CodingViewProps) {
             />
           </ButtonRow>
 
-          {!!paid ? markUnpaid : null}
-
+          {!paid ? markUnpaid : markUnpaid}
```
Comment (kevkevinpal) [inline frontend/app/src/people/widgetViews/summaries/wantedSummaries/CodingMobile.tsx:164]:
```suggestion
          { markPaidOrUnpaid }
```

**Comment by ecurrencyhodler** [comment]
Should we do a code review @gouravmpk ?

**Comment by gouravmpk** [comment]
Yeah sure

File: frontend/app/src/people/widgetViews/summaries/wantedSummaries/CodingMobile.tsx (hunk 1 above)
Comment (gouravmpk) [reply frontend/app/src/people/widgetViews/summaries/wantedSummaries/CodingMobile.tsx]: can we remove this line as its not needed ?

File: frontend/app/src/people/widgetViews/summaries/wantedSummaries/CodingMobile.tsx (hunk 3 above)
Comment (kevkevinpal) [reply frontend/app/src/people/widgetViews/summaries/wantedSummaries/CodingMobile.tsx:164]: could you address this comment @gouravmpk

Comment (gouravmpk) [reply frontend/app/src/people/widgetViews/summaries/wantedSummaries/CodingMobile.tsx:164]: Pushing

File: frontend/app/src/people/widgetViews/summaries/wantedSummaries/CodingMobile.tsx (hunk 1 above)
Comment (gouravmpk) [reply frontend/app/src/people/widgetViews/summaries/wantedSummaries/CodingMobile.tsx]: Done

File: frontend/app/src/people/widgetViews/summaries/wantedSummaries/CodingBounty.tsx [hunk 4]
```diff
@@ -398,31 +409,33 @@ function MobileView(props: CodingBountiesProps) {
         titleString={titleString}
         showPayBounty={showPayBounty}
         markUnpaid={
```
Comment (kevkevinpal) [inline frontend/app/src/people/widgetViews/summaries/wantedSummaries/CodingBounty.tsx:411]: I think you need to change that here aswell right? `markPaidOrUnpaid`

**Comment by gouravmpk** [comment]
@kevkevinpal check now

**Comment by kevkevinpal** [comment]
just noticing one issue with desktop

Should say mark unpaid
Screenshot 2024-01-08 at 9 34 33 AM

**Comment by kevkevinpal** [comment]
otherwise lgtm but since the ticket only worried about mobile I'll be merging now @gouravmpk in the future if we could add unit/component tests to the changes so we can insure it does not break in the future! Good work tho!

**Comment by ecurrencyhodler** [comment]
Tested and it works. GJ Gourav. Paid.

**Pull request merged**